### DISCUSSION TITLE
Source generators - invoke manually

### DISCUSSION THREAD

**Discussion by satyajit-behera** [dotnet/roslyn#52557] [Q&A]
How can we invoke source generator manually instead of only during compile time. If we want to generate some file during runtime using source generators, can we do that?

**Comment by ryzngard** [comment]
@satyajit-behera this is technically what the test infrastructure is doing. The main question I have is... why?

**Reply by satyajit-behera** [reply]
Was thinking if I could use this for email formatting and generating based on inputs.

**Comment by huoyaoyuan** [comment]
https://github.com/dotnet/runtime/blob/4d74f248c5c478b09ba7e78afd345947ccecc862/src/libraries/System.Text.Json/gen/unit_tests/CompilationHelper.cs#L57-L61

Found in runtime unit test.
